lib/uint64.js

```diff
@@ -0,0 +1,274 @@
+'use strict';
+
+const UINT32_MAX = 0xffffffff;
+
+const charToNum = {
+  0: 0, 1: 1, 2: 2,
+  3: 3, 4: 4, 5: 5,
+  6: 6, 7: 7, 8: 8,
+  9: 9,
+  a: 10, A: 10,
+  b: 11, B: 11,
+  c: 12, C: 12,
+  d: 13, D: 13,
+  e: 14, E: 14,
+  f: 15, F: 15,
+};
+
+class Uint64 {
+  constructor(value) {
+    this.value = new Uint32Array(2);
+    if (value instanceof Uint64) {
+      this.value[0] = value.value[0];
+      this.value[1] = value.value[1];
+      return;
+    }
+    const numValue = Number(value);
+    if (Number.isNaN(numValue) || numValue <= 0) {
+      return;
+    }
+    if (numValue <= UINT32_MAX) {
+      this.value[0] = numValue;
+    } else if (numValue <= Number.MAX_SAFE_INTEGER) {
+      this.value[0] = numValue % (UINT32_MAX + 1);
+      this.value[1] = Math.floor(numValue / (UINT32_MAX + 1));
+    } else if (typeof value === 'string') {
+      let radix = 10;
+      if (value.startsWith('0x')) {
+        radix = 16;
+      } else if (value.startsWith('0o')) {
+        radix = 8;
+      } else if (value.startsWith('0b')) {
+        radix = 2;
+      }
+      const uintRadix = new Uint64(radix);
+      let res = new Uint64();
+      for (let i = 0; i < value.length; i++) {
+        const digit = charToNum[value[i]];
+        const uintDigit = new Uint64(digit);
+        res = Uint64.mult(res, uintRadix).add(uintDigit);
+      }
+      this.value[0] = res.value[0];
+      this.value[1] = res.value[1];
+    }
+  }
+
+  static add(a, b) {
+    return new Uint64(a).add(b);
```

Comment:
I am not wrapping, I am creating a copy here, `a` must be a `Uint64` instance.

Reply:
So does this mean that the second argument doesn't have to be Uint64?

Reply:
All of the binary operations (except for bit-shifts) operate under the assumption that both arguments provided are `Uint64` values (bit-shifts expect `Uint64` as the first argument and `number` as the second).
I've just decided not to enforce this by throwing `TypeError` when the value of some other type is provided by the user, because it will probably slow down all of these operations, but we can discuss input validation here and maybe change it if needed.

Reply:
No, I'm talking about operations class-type operations (Uint64.add, Uint64.or etc) I think we should either 

* add type checking in them and allow to pass Uint64 and number as arguments
* add separate methods for Uint64-args and Uint64-number, number-number arguments.
* make them always expect Uint64 as arguments

I'm in favor of the first option as it will imo provide the best API. Also, we may pass `result` as last argument to specify whether to mutate first, create new or use the provided i.e. `undefined` -> create new (current behavior), `null` -> use first arg (mutate), `value` -> write result into `value`.